<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GrepWidget</class>
 <widget class="QWidget" name="GrepWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>730</width>
    <height>330</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>730</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Find-Replace In Files</string>
  </property>
  <layout class="QGridLayout">
   <item row="0" column="1" colspan="5">
    <widget class="KComboBox" name="patternCombo">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Enter the regular expression you want to search for here.</string>
     </property>
     <property name="whatsThis">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Enter the regular expression you want to search for here.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;If you do not check &quot;Regular Expression&quot; below, this is considered a raw string. That means, all meta characters are escaped.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Possible meta characters are:&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;.&lt;/span&gt; - Matches any character&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;^&lt;/span&gt; - Matches the beginning of a line&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;$&lt;/span&gt; - Matches the end of a line&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;\b&lt;/span&gt; - Matches a word boundary&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;\s&lt;/span&gt; - Matches any whitespace character&lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;The following repetition operators exist:&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;?&lt;/span&gt; - The preceding item is matched at most once&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;*&lt;/span&gt; - The preceding item is matched zero or more times&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;+&lt;/span&gt; - The preceding item is matched one or more times&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;{&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;n&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;}&lt;/span&gt; - The preceding item is matched exactly &lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt; times&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;{&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;n&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;,}&lt;/span&gt; - The preceding item is matched &lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt; or more times&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;{,&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;n&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;}&lt;/span&gt; - The preceding item is matched at most &lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt; times&lt;/li&gt;
&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;{&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;n&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;,&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;m&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;}&lt;/span&gt; - The preceding item is matched at least &lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt;, but at most &lt;span style=&quot; font-style:italic;&quot;&gt;m&lt;/span&gt; times.&lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Furthermore, backreferences to bracketed subexpressions are available via the notation \&lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt;.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;For further reference, look at &lt;a href=&quot;http://www.pcre.org&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;www.pcre.org&lt;/span&gt;&lt;/a&gt; or &lt;span style=&quot; font-style:italic;&quot;&gt;man pcresyntax.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="autoCompletion">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>Template:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>templateEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="3">
    <widget class="KComboBox" name="templateEdit">
     <property name="toolTip">
      <string>This is the regular expression template.</string>
     </property>
     <property name="whatsThis">
      <string>This is the regular expression template. &lt;i&gt;%s&lt;/i&gt; will be replaced by the pattern, while &lt;i&gt;%%&lt;/i&gt; will be replaced by &lt;i&gt;%&lt;/i&gt;.</string>
     </property>
    </widget>
   </item>
   <item row="5" column="5">
    <widget class="KPushButton" name="syncButton">
     <property name="toolTip">
      <string>Quickly select location from a set of directories</string>
     </property>
     <property name="text">
      <string>Quick Select</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="depthLabel">
     <property name="text">
      <string>Depth:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>depthSpin</cstring>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QSpinBox" name="depthSpin">
     <property name="toolTip">
      <string extracomment="Maximum level of recursion. For folder-based searches, that means the level of subfolders. Choose 0 to look only in files inside the current folder, or Full to search also on the subfoldersâ€™s files, theirs subfoldersâ€™ files, and so on."/>
     </property>
     <property name="specialValueText">
      <string extracomment="Full recursion will be used. For folder-based searches, that means searching in the target folder and all the subfolders, their subfolders, and so on.">Full</string>
     </property>
     <property name="minimum">
      <number>-1</number>
     </property>
     <property name="value">
      <number>-1</number>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>Pattern:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>patternCombo</cstring>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Files:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>filesCombo</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QLabel" name="label_11">
     <property name="text">
      <string>Case Sensitive:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>caseSensitiveCheck</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="3">
    <widget class="QCheckBox" name="caseSensitiveCheck">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Case-sensitive searching.</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Location(s):</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>searchPaths</cstring>
     </property>
    </widget>
   </item>
   <item row="9" column="3" colspan="3">
    <spacer>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>361</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="4" colspan="2">
    <widget class="KComboBox" name="templateTypeCombo"/>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>Regular Expression:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>regexCheck</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QCheckBox" name="regexCheck">
     <property name="toolTip">
      <string>Enable or disable regular expression.</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="6" column="2">
    <widget class="QLabel" name="limitToProjectLabel">
     <property name="text">
      <string>Limit to project files:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>limitToProjectCheck</cstring>
     </property>
    </widget>
   </item>
   <item row="6" column="3">
    <widget class="QCheckBox" name="limitToProjectCheck">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="7" column="1" colspan="5">
    <widget class="KComboBox" name="filesCombo">
     <property name="toolTip">
      <string>Files filter.</string>
     </property>
     <property name="whatsThis">
      <string>Enter the file name pattern of the files to search here. You may give several patterns separated by commas or spaces.</string>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="autoCompletion">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Exclude:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>excludeCombo</cstring>
     </property>
    </widget>
   </item>
   <item row="8" column="1" colspan="5">
    <widget class="KComboBox" name="excludeCombo">
     <property name="toolTip">
      <string>Files pattern to exclude</string>
     </property>
     <property name="whatsThis">
      <string>Enter the file name pattern of the files to exclude from the search here. You may give several patterns separated by commas or spaces.&lt;p&gt;Every pattern is internally surrounded by asterisks, so that each pattern can match parts of the file paths.&lt;/p&gt;</string>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="autoCompletion">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <spacer>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>141</width>
       <height>16</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="9" column="1">
    <spacer>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>41</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="9" column="2">
    <spacer>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>151</width>
       <height>31</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="1" colspan="3">
    <widget class="KComboBox" name="replacementTemplateEdit">
     <property name="toolTip">
      <string>Enter the replacement template.</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Replacement Template:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>replacementTemplateEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="1" colspan="3">
    <widget class="KComboBox" name="searchPaths">
     <property name="toolTip">
      <string>Select the location where you want to search. It can be a directory, a file, or a semicolon separated (without spaces) list of directories/files.</string>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="autoCompletion">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="4">
    <widget class="QPushButton" name="directorySelector">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Select a directory to search in.</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KPushButton</class>
   <extends>QPushButton</extends>
   <header>kpushbutton.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>patternCombo</tabstop>
  <tabstop>templateEdit</tabstop>
  <tabstop>templateTypeCombo</tabstop>
  <tabstop>replacementTemplateEdit</tabstop>
  <tabstop>regexCheck</tabstop>
  <tabstop>caseSensitiveCheck</tabstop>
  <tabstop>searchPaths</tabstop>
  <tabstop>directorySelector</tabstop>
  <tabstop>syncButton</tabstop>
  <tabstop>depthSpin</tabstop>
  <tabstop>limitToProjectCheck</tabstop>
  <tabstop>filesCombo</tabstop>
  <tabstop>excludeCombo</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
